geniza/corpus/metadata_export.py

```diff
@@ -0,0 +1,239 @@
+    outd["tags"] = doc.all_tags()
+    outd["description"] = doc.description
+    outd["shelfmarks_historic"] = sep_within_cells.join(
+        [os for os in old_shelfmarks if os]
+    )
+    outd["languages_primary"] = doc.all_languages()
+    outd["languages_secondary"] = doc.all_secondary_languages()
+    outd["language_note"] = doc.language_note
+    outd["doc_date_original"] = doc.doc_date_original
+    outd["doc_date_calendar"] = doc.doc_date_calendar
+    outd["doc_date_standard"] = doc.doc_date_standard
+    outd["notes"] = doc.notes
+    outd["needs_review"] = doc.needs_review
+    outd[
+        "url_admin"
+    ] = f"{url_scheme}{site_domain}/admin/corpus/document/{doc.id}/change/"
+    outd["initial_entry"] = (
+        all_log_entries.last().action_time if all_log_entries else ""
+    )
+    outd["last_modified"] = doc.last_modified
+    outd["input_by"] = sep_within_cells.join(
+        sorted(
+            list(set([user.get_full_name() or user.username for user in input_users]))
+        )
+    )  # sorting to ensure deterministic order
+    outd["status"] = doc.get_status_display()
+    outd["library"] = sep_within_cells.join(libraries) if any(libraries) else ""
+    outd["collection"] = sep_within_cells.join(collections) if any(collections) else ""
+
+    return outd
+
+
+def iter_export_data_for_docs(
+    docs=None,
+    progress=CSV_EXPORT_PROGRESS,
+    sep_within_cells=SEP_WITHIN_CELLS,
+):
+    timeprint("iter_export_data_for_docs()")
+
+    # get docs
+    if not docs:
+        docs = get_docs_to_export()
+
+    # progress bar?
+    timeprint("Beginning to write")
+    iterr = docs if not progress else track(docs, desc=f"Writing rows to file")
+
+    # save
+    for doc in iterr:
+        yield get_export_data_for_doc(doc, sep_within_cells=sep_within_cells)
+
+
+def iter_export_data_for_docs_as_ll(
+    docs=None, csv_fields=CSV_FIELDS, progress=CSV_EXPORT_PROGRESS, default="", **kwargs
+):
+    timeprint("iter_export_data_for_docs_as_ll()")
+    iterr = iter_export_data_for_docs(docs=docs, progress=progress, **kwargs)
+    for docd in iterr:
+        yield [docd.get(h, default) for h in csv_fields]
+
+
+def stream_export_data_for_docs(
+    fn="pgp_documents.csv",
+    docs=None,
+    pseudo_buffer=False,
+    csv_fields=CSV_FIELDS,
+    **kwargs,
+):
+    timeprint("stream_export_data_for_docs()")
+    with (open(fn, "w") if not pseudo_buffer else Echo()) as of:
+        writer = csv.DictWriter(of, fieldnames=csv_fields, extrasaction="ignore")
+        yield writer.writeheader()
+        for docd in iter_export_data_for_docs(docs=docs):
+            yield writer.writerow(docd)
+
+
+def write_stream_export_data_for_docs(
+    fn="pgp_documents.csv", docs=None, csv_fields=CSV_FIELDS, **kwargs
+):
+    timeprint("write_stream_export_data_for_docs()")
+    for row in stream_export_data_for_docs(
+        fn=fn, docs=docs, csv_fields=csv_fields, **kwargs
+    ):
+        pass
+
+
+def http_stream_export_data_for_docs(
```

Comment:
as we discussed : let's rename export methods so they are not document specific, and we can reuse the same approach with all our other exports; I guess they should take a queryset and a filename; what else? does it need to take a metadata generator instead of a queryset?